docs/en/pact-reference.md

```diff
@@ -19,265 +19,796 @@ blockchain applications with just the `pact` tool.
 To start up the server issue `pact -s config.yaml`, with a suitable config.
 The `pact-lang-api` JS library is [available via npm](https://www.npmjs.com/package/pact-lang-api) for web development.
 
+### Commands
 
-`cmd` field and Payloads {#cmd-field-and-payloads}
----
+#### The command object {#the-command-object}
+
+Pact represents blockchain transactions (or commands) as a JSON object with the
+following attributes:
+
+##### Attributes
+
+###### `"cmd"`
+_type:_ **string (encoded, escaped JSON)** `required`
+
+The component of a transaction that is signed to ensure non-malleability.
+Initially, this component is a JSON object containing the transaction payload,
+signatories, and platform-specific metadata. When assembling the transaction,
+this JSON is "stringified" and provided to the `cmd` field.
+If you inspect the output of the
+[request formatter in the pact tool](#api-request-formatter),
+you will see that the `"cmd"` field, along with any code supplied, are a
+String of encoded, escaped JSON. See
+[cmd field](#cmd-field) for more information.
+
+###### `"hash"`
+_type:_ **string (base64url)** `required`
+
+Blake2 hash of the `cmd` field "stringified" value. Serves as a command's
+request key since each transaction must be unique.
+
+###### `"sigs"`
+_type:_ **[ object ]** `required`
+
+List of JSON objects containing a cryptographic signature. This signature
+is generated using a cryptographic private key to authenticate
+the current transaction (i.e. the contents of the `cmd` field). The `ith`
+signature must correspond to the `ith` signer in `cmd`'s list of
+[signers](#cmd-signers).This signature object has the following attribute(s):
+
+```yaml
+name: "sig"                     # Cryptographic signature of current
+type: string (base16)           # transaction.
+required: true    
+```
+
+##### Example command
+```JSON
+// Command with exec payload and Chainweb metadata
+{
+  "cmds":[{
+    "hash":"H6XjdPHzMai2HLa3_yVkXfkFYMgA0bGfsB0kOsHAMuI",
+    "sigs":[{
+      "sig":"8d452109cc0439234c093b5e204a7428bc0a54f22704402492e027aaa9375a34c910d8a468a12746d0d29e9353f4a3fbebe920d63bcc7963853995db015d060f"
```

Comment:
wow, our sigs are still this big? and not base64url encoded either ...

Reply:
Should I make signatures base64url in this PR?

Reply:
No -- just checking that this is what they are now

Reply:
oh gotcha! yes the signatures are still this long